relay-common/src/metrics.rs

```diff
@@ -64,119 +125,328 @@ where
     })
 }
 
+/// A metric for capturing timings.
+///
+/// Timings are a positive number of milliseconds between a start and end time. Examples include
+/// time taken to render a web page or time taken for a database call to return.
+///
+/// ## Example
+///
+/// ```
+/// use relay_common::{metric, metrics::TimerMetric};
+///
+/// enum MyTimer {
+///     ProcessA,
+///     ProcessB,
+/// }
+///
+/// impl TimerMetric for MyTimer {
+///     fn name(&self) -> &'static str {
+///         match self {
+///             Self::ProcessA => "process_a",
+///             Self::ProcessB => "process_b",
+///         }
+///     }
+/// }
+///
+/// # fn process_a() {}
+///
+/// // measure time by explicitly setting a std::timer::Duration
+/// # use std::time::Instant;
+/// let start_time = Instant::now();
+/// process_a();
+/// metric!(timer(MyTimer::ProcessA) = start_time.elapsed());
+///
+/// // provide tags to a timer
+/// metric!(
+///     timer(MyTimer::ProcessA) = start_time.elapsed(),
+///     server = "server1",
+///     host = "host1",
+/// );
+///
+/// // measure time implicitly by enclosing a code block in a metric
+/// metric!(timer(MyTimer::ProcessA), {
+///     process_a();
+/// });
+///
+/// // measure block and also provide tags
+/// metric!(
+///     timer(MyTimer::ProcessB),
+///     server = "server1",
+///     host = "host1",
+///     {
+///         process_a();
+///     }
+/// );
+///
+/// // using a raw timer (directly sending a raw value)
+/// let raw_duration = start_time.elapsed().as_millis() as u64;
+/// metric!(
+///     time_raw(MyTimer::ProcessA) = raw_duration,
```

Comment:
@RaduW What do you think about introducing a crate specifically for raw timers. 

We use those for recording individual raw values. In this case, suche as payload sizes or entity counts within requests. Internally, we'd still use the timer, but we could do essentially this:

```rust 
pub trait ValueMetric { /* ... */ }

metric!(value(RelayValues::EventSize) = event.json_size() as i64);
```

Reply:
Using raw timers for non timing related values is wrong, for that we should use histograms for that.
So maybe we should just remove timer raw (if that is what it is used for).